MetaMask/metamask-mobile
# feat: 2805 grant permission to network with missmatching rpc url

## **Description**
Behind feature flag for now, adds the ability to see a comparison of the missmatched rpc URL when it differs from the expected rpc URL. These are UI changes only for now, and will require controller work to do the rpc URL of the new network permission being added, with expected current excpected URL.


## **Related issues**

Contributes to resolving: https://github.com/MetaMask/MetaMask-planning/issues/2805

## **Manual testing steps**

1. Connect one or more account(s) to a dapp in the in app browser
2. Select the dapp that offers an unexpected network URL for a popular network, in the dapp choose a popular network like polygon
3. This will create a network switching and the bottom sheet will show up, with a blue tag, clickable, click it to see the details of the current and new URL.

## **Screenshots/Recordings**



| Before       | After        |
|--------------|--------------|
| <img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/38e6618d-0724-4917-8f7e-106cb6b23884"> |<img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/2ea4f118-76cf-43fd-b9ad-e81825cd7e9b"> |



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
